autoload/highlightedyank.vim

```diff
@@ -85,6 +85,55 @@ function! s:yank_normal(count, register) abort "{{{
   endif
 endfunction
 "}}}
+function! highlightedyank#autocmd_highlight() abort "{{{
+  let view = winsaveview()
+  let region = {}
+  let region.head = getpos("'[")
+  let region.tail = getpos("']")
+  let char = getline(region.tail[1])[region.tail[2] - 1]
+  let linecontent = v:event.regcontents[len(v:event.regcontents) - 1]
+  let tailchar = linecontent[len(linecontent) - 1]
+  if char != tailchar
+    let region.tail[2] = region.tail[2] - 1
+  endif
+  if v:event.regtype == 'v'
+    let motionwise = 'char'
+  elseif v:event.regtype == 'V'
+    let motionwise = 'line'
+  elseif v:event.regtype =~ "\<c-v>"
+    let motionwise = 'block'
+  else
+    let motionwise = ''
+  endif
```

Comment:
Refactored. However, for "<c-v>" I still kept the conditional, as the motionwise also included the numeric width, as specified in the [docs](https://neovim.io/doc/user/eval.html#getregtype()).

The underlying highlight function didn't work with `CTRL-V100`, for example.